napari/layers/labels/_labels_key_bindings.py

```diff
@@ -90,6 +90,13 @@ def increase_label_id(layer: Labels):
     layer.selected_label += 1
 
 
+@register_label_action(
+    trans._("Toggle preserve labels"),
+)
```

Comment:
```suggestion
@register_layer_attr_action(Labels, trans._("Toggle preserve labels"), "preserve_labels")
```

Fixed version if you prefere toggle. 
I forgot about `Labels` argument.

Reply:
Bingo. [Google didn't bring me any help with this decorator, is it not in docs or just not in docs that have been spidered.]
Not sure how important the on-hold, because it's a binary checkbox, but it does make the behavior consistent with the old keybind, `Shift`, even if that keybind isn't settable at the moment (https://github.com/napari/napari/issues/5016#issuecomment-1237416747)

Reply:
Example of usage is here: https://github.com/napari/napari/blob/main/napari/layers/labels/_labels_key_bindings.py#L31

and definition here:

https://github.com/napari/napari/blob/main/napari/layers/utils/layer_utils.py#L66

It is a pure napari thing, so it may be hard to google. 



> Not sure how important the on-hold, because it's a binary checkbox, but it does make the behavior consistent with the old keybind, `Shift`, even if that keybind isn't settable at the moment ([#5016 (comment)](https://github.com/napari/napari/issues/5016#issuecomment-1237416747))

I will try to fix this in #5018

Reply:
That would rock.
Combined with alternate keybinds, one could bind `P` (which is close to -/= used to increment) and `Shift` which is just generally easy via pinky to hit.

Reply:
> That would rock.

implemented in PR